script.js

```diff
@@ -1,28 +1,276 @@
-  }
-  function getOutput() {
+}
+function getOutput() {
     return document.getElementById("output-value").innerText;
-  }
-  function printOutput(num) {
+}
+function printOutput(num) {
     if (num == "") {
-      document.getElementById("output-value").innerText = num;
+        document.getElementById("output-value").innerText = num;
     } else {
-      document.getElementById("output-value").innerText = getFormat(num);
+        document.getElementById("output-value").innerText = getFormat(num);
     }
-  }
-  function getFormat(num) {
+}
+function getFormat(num) {
     if (num == "-") {
-      return "";
+        return "";
     }
     var n = Number(num);
     var value = n.toLocaleString("en");
     return value;
-  }
-  function reverseNumberFormat(num) {
+}
+function reverseNumberFormat(num) {
     return Number(num.replace(/,/g, ""));
-  }
-  
\ No newline at end of file
+}
+
+// All operators
+var operator = document.getElementsByClassName("operator");
+for (var i = 0; i < operator.length; i++) {
+    operator[i].addEventListener("click", function () {
+        if (this.id == "clear") {
+            printHistory("");
+            printOutput("");
+        } else if (this.id == "backspace") {
+            var output = reverseNumberFormat(getOutput()).toString();
+            if (output) {
+                //if output has a value
+                output = output.substring(0, output.length - 1);
+                printOutput(output);
+            }
+        } else {
+            var output = getOutput();
+            var history = getHistory();
+            if (output != "" || output == "" || history != "") {
+                output = output == "" ? output : reverseNumberFormat(output);
+                history = history + output;
+                if (this.id == "=") {
+                    var result = eval(history);
+                    printOutput(result);
+                    printHistory("");
+                } else {
+                    history = history + this.id;
+                    printHistory(history);
+                    printOutput("");
+                }
+            }
+        }
+    });
+}
+
+// All numbers
+
+var number = document.getElementsByClassName("number");
+for (var i = 0; i < number.length; i++) {
+    number[i].addEventListener("click", function () {
+        var output = reverseNumberFormat(getOutput());
+        if (output != NaN) {
+            //if output is a number
+            output = output + this.id;
+            printOutput(output);
+        }
+    });
+}
+
+//Trigonometry Functions
+
+function sin() {
+    document.getElementById("output-value").innerText = Math.sin(
+        document.getElementById("output-value").innerText
+    );
+}
+
+function cos() {
+    document.getElementById("output-value").innerText = Math.cos(
+        document.getElementById("output-value").innerText
+    );
+}
+
+function sin() {
+    document.getElementById("output-value").innerText = Math.tan(
```

Comment:
Instead of repeatedly querying the document for `output-value`, create a global variable which holds the same value.